ui/core/proto_utils/equipped_item.ts

```diff
@@ -1,14 +1,42 @@
-import { GemColor, ItemRandomSuffix, ItemSpec, ItemType, Profession, ReforgeStat, Stat } from '../proto/common.js';
+import {
+	GemColor,
+	ItemLevelState,
+	ItemRandomSuffix,
+	ItemSlot,
+	ItemSpec,
+	ItemType,
+	Profession,
+	PseudoStat,
+	ReforgeStat,
+	ScalingItemProperties,
+	Stat,
+} from '../proto/common.js';
 import { UIEnchant as Enchant, UIGem as Gem, UIItem as Item } from '../proto/ui.js';
 import { distinct } from '../utils.js';
 import { ActionId } from './action_id.js';
 import { gemEligibleForSocket, gemMatchesSocket } from './gems.js';
 import { Stats } from './stats.js';
 import { enchantAppliesToItem } from './utils.js';
 
-export function getWeaponDPS(item: Item): number {
-	return (item.weaponDamageMin + item.weaponDamageMax) / 2 / (item.weaponSpeed || 1);
-}
+export const getWeaponDPS = (item: Item, upgradeStep: ItemLevelState = ItemLevelState.Base): number => {
+	const { weaponDamageMin, weaponDamageMax } = item.scalingOptions[upgradeStep];
+	return (weaponDamageMin + weaponDamageMax) / 2 / (item.weaponSpeed || 1);
+};
+
+export const getWeaponStatsBySlot = (item: Item, slot: ItemSlot) => {
+	let itemStats = new Stats();
```

Comment:
Shouldn't this be
```
let itemStats = new Stats(item.stats);
```
Unless the intent is to compute only the weapon PseudoStat contribution, in which case the output of this function should be _added_ to the base item stats rather than replacing them.